### ISSUE TITLE
Arr::map doesn't work with multidimensional array and keys 

### ISSUE THREAD

**Issue by acidtv** [kohana/core#623]
The current implementation of Arr::map() does not pass the $keys array on to itself when doing the recursive call. The result is that $callbacks is applied to every element in sub-arrays, not just to the fields specified in $keys.

For example:

``` php
Arr::map('strip_tags', array(array('foo' => '<p>bar</p>', 'bar' => '<p>baz</p>')), 'foo')
```

Returns

```
array
  0 => 
    array
      'foo' => string 'bar' (length=3)
      'bar' => string 'baz' (length=3)
```

The 'bar' element here should have been untouched.

I can issue a pull request and add unit tests for this if someone tells me in which branch this can best be fixed :)

**Comment by enov** [comment]
Hey @acidtv, thanks for this report.

Bug fixes should be targeted to `3.3/develop` whenever possible. Gets usually merged up to `3.4/develop` later.

**Comment by acidtv** [comment]
Ok, I've issued a pull request.
